### ISSUE TITLE
Non-modal dialog in the background closes the active modal dialog and blocks user interaction

### ISSUE THREAD

**Issue by winson1209** [Caliburn-Micro/Caliburn.Micro#440]
I have a non-modal loading dialog running in the background with a background worker, but at the same time a separate modal dialog pops up while it's "loading". When the loading finishes, it closes the loading dialog fine, but it also incorrectly closes the modal dialog. However, the program doesn't seem to know that the modal dialog is now closed and is stuck waiting for user input with the modal dialog. As a result, any further interaction with the program is blocked by the modal dialog that doesn't exist anymore.

I have attached a very simple project that demonstrate this problem:
[ConflictingDialogs.zip](https://github.com/Caliburn-Micro/Caliburn.Micro/files/1019538/ConflictingDialogs.zip)

Many thanks in advance.

**Comment by nigel-sampson** [comment]
It appear what's happening is that because the Window in the sample is being open it's the active window. The WindowManager then infers the ownership of the dialog to that Window. Closing the window then closes the dialog.

IMO this makes sense, although there seems to platform bug around closing the window when it has a dialog, but that's out of scope for Caliburn.Micro.

**Comment by winson1209** [comment]
Thank you for your reply.

Is there any way to specify which window is the owner of the active window? Or is there any workaround that you would suggest?

**Comment by winson1209** [comment]
Thank you very much. I will give that a go.
